include/red_black_tree.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2019, DSA authors.
+ * All rights reserved.
+ * Authors: Allen
+ */
+
+// 红黑树
+#pragma once
+
+#include <memory>
+#include <queue>
+#include <vector>
+#include <optional>
+#include <iostream>
+
+#include <binary_search_tree.h>
```

Comment:
Let use `#include "binary_search_tree.h` since the header file is in our project.